astropy/stats/lombscargle/core.py

```diff
@@ -356,3 +356,97 @@ def model(self, t, frequency):
                              fit_mean=self.fit_mean,
                              nterms=self.nterms)
         return y_fit * get_unit(self.y)
+
+    def cumulative_distribution(self, z, normalization='standard'):
+        """
+        Compute the theoretical cumulative distribution of the periodogram
+        for the null hypothesis – i.e. data consisting of Gaussian noise.
+
+        Parameters
+        ----------
+        z : array-like
+            the periodogram value
+        normalziation : string
+            The periodogram normalization. Must be one of
+            ['standard', 'model', 'log', 'psd']
+
+        Returns
+        -------
+        cdf : np.ndarray
+            The theoretical cumulative distribution function
+
+        Notes
+        -----
+        Expressions used here are adapted from Baluev 2008, Table 1.
+        """
+        # TODO: handle units in z?
+        z = np.asarray(z)
```

Comment:
This will needlessly break for someone putting in `z` that is has a scale but is otherwise dimensionless (say, percent). My sense would be not to worry about this for now, i.e. I'd remove this statement and its comment, as anything with a true dimension will break below anyway and we can see if it makes sense to address that if someone complains.